File: trendly_flutter/lib/models/bookmark.dart
class Bookmark {
  final int id;
  final int userId;
  final String articleUrl;
  final String title;
  final String sourceName;
  final String imageUrl;

  Bookmark({
    required this.id,
    required this.userId,
    required this.articleUrl,
    required this.title,
    required this.sourceName,
    required this.imageUrl,
  });

  factory Bookmark.fromJson(Map<String, dynamic> json) {
    return Bookmark(
      id: json['id'] is int ? json['id'] : int.parse(json['id'].toString()),
      userId: json['userId'] is int
          ? json['userId']
          : int.parse(json['userId'].toString()),
      articleUrl: json['articleUrl'] ?? '',
      title: json['title'] ?? '',
      sourceName: json['sourceName'] ?? '',
      imageUrl: json['imageUrl'] ?? '',
    );
  }
}

File: trendly-backend/src/main/java/com/trendly/trendlybackend/bookmark/model/Bookmark.java
package com.trendly.trendlybackend.bookmark.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(
    name = "bookmarks",
    uniqueConstraints = @UniqueConstraint(columnNames = {"user_id", "article_url"})
)
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Bookmark {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    // Which user saved this article
    @Column(name = "user_id", nullable = false)
    private Long userId;

    // URL of the news article
    @Column(name = "article_url", nullable = false, length = 1000)
    private String articleUrl;

    // Optional: store title for quicker display
    @Column(name = "title", length = 500)
    private String title;

    // Source like "BBC News", "NDTV"
    @Column(name = "source_name")
    private String sourceName;

    // Optional image URL
    @Column(name = "image_url", length = 1000)
    private String imageUrl;

    private LocalDateTime createdAt;
}
